src/components/DataSession.vue

```diff
@@ -12,43 +22,15 @@ const props = defineProps({
 	}
 })
 
-function addOperation(operationDefinition) {
-	// Create data operation in this session with input parameters specified from wizard
-	fetch('http://127.0.0.1:8000/api/datasessions/' + props.data.id + '/operations/', {
-		'method': 'POST',
-		'body': JSON.stringify(operationDefinition),
-		'headers': {
-			'Authorization': 'Token 123456789abcdefg',
-			'Content-type': 'application/json; charset=UTF-8'
-		}
-	})
-		.then(response => response.json())
-		.then(function() {emit('reloadSession')})
-		.catch(error => console.error('Error:', error))
+async function addOperation(operationDefinition) {
+	const url = dataSessionsUrl + props.data.id + '/operations/'
+	await fetchApiCall({url: url, method: 'POST', body: operationDefinition, headers: authHeaders, successCallback: emit('reloadSession'), failCallback: handleError})
 }
 
 const getImages = async () => {
-	try {
-		const response = await fetch ('http://127.0.0.1:8000/api/datasessions/' + props.data.id, {
-			method: 'GET',
-			headers: {
-				'Authorization': 'Token 123456789abcdefg',
-				'Content-Type': 'application/json; charset=UTF-8'
-			}
-		})
-
-		if (!response.ok) {
-			const errorData = await response.json()
-			console.error('Error Response Data:', errorData)
-			throw new Error('Bad request')
-		}
-
-		const data = await response.json()
-		images.value = data.input_data
-	} catch (error) {
-		console.log('Error getting images: ', error)
-	}
-
+	const url = dataSessionsUrl + props.data.id
+	const data = await fetchApiCall({url: url, method: 'GET', headers: authHeaders, failCallback: handleError})
```

Comment:
I think the `successCallback` here could be something as simple as `images.value = data.input_data`

Reply:
I thought passing an assignment as a function argument is a big no no, but I might be wrong. could make a function that just has that as a one liner. Wasn't sure the best way to implement these so I'd be open to discussion.

Reply:
I'd just define it as an inline arrow function for now - happy to talk about a refactor later on if we need to.